aptos-move/e2e-move-tests/src/tests/aggregator_v2_events.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    assert_success,
+    tests::{aggregator_v2::EAGGREGATOR_OVERFLOW, common},
+    MoveHarness,
+};
+use aptos_language_e2e_tests::account::Account;
+use aptos_types::{
+    on_chain_config::FeatureFlag,
+    transaction::{ExecutionStatus, SignedTransaction, TransactionOutput},
+};
+use claims::{assert_matches, assert_ok, assert_some};
+use move_core_types::{
+    account_address::AccountAddress, ident_str, language_storage::ModuleId,
+    parser::parse_struct_tag, vm_status::AbortLocation,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize)]
+struct Aggregator {
+    value: u64,
+    max_value: u64,
+}
+
+#[derive(Deserialize, Serialize)]
+struct AggregatorSnapshot {
+    value: u64,
+}
+
+#[derive(Deserialize, Serialize)]
+struct Counter {
+    value: Aggregator,
+}
+
+#[derive(Deserialize, Serialize)]
+struct Event {
+    value: AggregatorSnapshot,
+}
+
+fn publish_test_package(h: &mut MoveHarness, aptos_framework_account: &Account) {
+    let path_buf = common::test_dir_path("aggregator_v2.data/pack");
+    assert_success!(h.publish_package_cache_building(aptos_framework_account, path_buf.as_path()));
+}
+
+fn create_test_txn(
+    h: &mut MoveHarness,
+    aptos_framework_account: &Account,
+    name: &str,
+) -> SignedTransaction {
+    h.create_entry_function(
+        aptos_framework_account,
+        str::parse(name).unwrap(),
+        vec![],
+        vec![bcs::to_bytes(aptos_framework_account.address()).unwrap()],
+    )
+}
+
+fn run_entry_functions<F: Fn(ExecutionStatus)>(
+    h: &mut MoveHarness,
+    aptos_framework_account: &Account,
+    func_names: Vec<&str>,
+    check_status: F,
+) -> Vec<TransactionOutput> {
+    publish_test_package(h, aptos_framework_account);
+
+    // Make sure aggregators are enabled, so that we can test.
+    h.enable_features(
+        vec![
+            FeatureFlag::AGGREGATOR_V2_API,
+            FeatureFlag::AGGREGATOR_V2_DELAYED_FIELDS,
+            FeatureFlag::RESOURCE_GROUPS_CHARGE_AS_SIZE_SUM,
+        ],
+        vec![],
+    );
+
+    let txns = func_names
+        .into_iter()
+        .map(|name| create_test_txn(h, aptos_framework_account, name))
+        .collect();
+
+    let outputs = h.run_block_get_output(txns);
+    for output in outputs.iter() {
+        let execution_status = assert_ok!(output.status().as_kept_status());
+        check_status(execution_status);
+    }
+    outputs
+}
+
+fn check_events(outputs: Vec<TransactionOutput>) {
+    // We always emit fee statement event. Currently this is hardcoded
+    // for existing test case.
+    assert_eq!(outputs[0].events().len(), 1);
+    assert_eq!(outputs[1].events().len(), 1);
+
+    assert_eq!(outputs[2].events().len(), 2);
+    let event = assert_ok!(bcs::from_bytes::<Event>(
```

Comment:
Should this be `bcs::from_bytes::<EventV1>` or `bcs::from_bytes::<EventV2>`?

Reply:
Both EventV1 and EventV2 have the same structure, so I deserialize them into the same Rust struct